lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
+  #to download the csv and save the query
+  check_done: ->
+    #check if all other queries have returned, if so can get total csv
+    row_arr = []
+    tab = $("#queryTableBody")
+    rows = tab.find("tr")
+    _.each rows, (row) ->
+      row_arr.push(row.getAttribute('query'))
+    allGood = true
+    _.each row_arr, (status) ->
+      if status == "working"
+        allGood = false
+    if allGood
+      @$save_query_btn.removeClass("disabled")
+      @$email_csv_btn.removeClass("disabled")
+      @$email_csv_btn[0].value = "Download CSV"
+
+  #deletes an active query from the table and the db
+  delete_temporary:->
+    queriesToDelete = []
+    _.each $("#queryTableBody tr"), (row) ->
+      if row.hasAttribute('query')
+        queryToDelete = row.getAttribute('query')
+        queriesToDelete.push(queryToDelete)
+    @delete_bulk_temp_query(queriesToDelete)
+    $("#queryTableBody tr").remove()
+
+  #adds a row to saved queries
+  start_saved_row:(color, arr, id, table) ->
+    #find which row to insert in
+    rows = table[0].children
+    row = table[0].insertRow(rows)
+    row.setAttribute("groupQuery", id)
+    for num in [0..1]
+      cell = row.insertCell(num)
+      item = arr[num]
+      cell.innerHTML = item['text']
+      if item.hasOwnProperty('id')
+        cell.id = item['id']
+    $load_btn = $ _.template('<div class="loadQuery"><i class="icon-upload">
+      </i> <%= label %></div>', {label: "Load"})
+    $load_btn.click =>
+      @delete_temporary()
+      $("#queryTableBody tr").remove()
+      targ = event.target
+      while (!targ.classList.contains('loadQuery'))
+        targ = targ.parentNode
+      curRow = targ.parentNode.parentNode
+      groupedQueryId = curRow.getAttribute('groupQuery')
+      @$email_csv_btn[0].value = "Aggregating Queries"
+      $("#incompleteMessage")[0].innerHTML = ""
+      rowsToAdd = $(".saved" + groupedQueryId)
+      for row in rowsToAdd
+        cells = row.children
+        @arr = [{'text':cells[0].innerText},
+                {'text':cells[1].innerText},
+                {'text':cells[2].innerText, 'id':cells[2].id},
+                {'text':cells[3].innerText}]
+
+        @arr_text = [cells[0].innerText, cells[1].innerText,
+                     cells[2].innerText, cells[3].innerText]
+        @tr = @start_row(cells[0].innerText.toLowerCase(),
+          @arr,"", $("#queryTableBody"))
+        #todo:this feels too hacky. suggestions?
+        @use_query_endpoint =@$query_endpoint + "/" + @arr_text.slice(0,2).
+          join("/") + "/" + @arr[2].id
+        @filtering = @arr[3].text
+        @entityName = @arr[2].text
+        @reload_students(@tr)
+        @$email_list_containers.find('select.single-email-selector').
+          prop('selectedIndex',0)
+        $(".problem_specific").removeClass('active')
+        $(".section_specific").removeClass('active')
+    $td = $ '<td>'
+    $td.append $load_btn
+    row.appendChild $td[0]
```

Comment:
we need to subscript it
$td is [<td><div class="deleteSaved">      <i class="icon-remove-sign"></i> Delete</div></td>]